core/shared/src/main/scala/scalaz/zio/UIO.scala

```diff
@@ -0,0 +1,534 @@
+   */
+  final def lock[A](executor: Executor)(zio: ZIO[UioR, UioE, A]): ZIO[UioR, UioE, A] =
+    ZIO.lock(executor)(zio)
+
+  /** Imports an asynchronous effect into a pure `ZIO` value. See `effectAsyncMaybe` for
+   * the more expressive variant of this function that can return a value
+   * synchronously.
+   */
+  final def effectAsync[A](register: (ZIO[Any, UioE, A] => Unit) => Unit): ZIO[Any, UioE, A] =
+    ZIO.effectAsync(register)
+
+  /** Imports an asynchronous effect into a pure `ZIO` value, possibly returning
+   * the value synchronously.
+   */
+  final def effectAsyncMaybe[A](register: (ZIO[Any, UioE, A] => Unit) => Option[IO[UioE, A]]): ZIO[Any, UioE, A] =
+    ZIO.effectAsyncMaybe(register)
+
+  /** Imports an asynchronous effect into a pure `ZIO` value. This formulation is
+   * necessary when the effect is itself expressed in terms of `ZIO`.
+   */
+  final def effectAsyncM[A](register: (ZIO[UioR, UioE, A] => Unit) => ZIO[UioR, Nothing, _]): ZIO[UioR, UioE, A] =
+    ZIO.effectAsyncM(register)
+
+  /** Imports an asynchronous effect into a pure `IO` value. The effect has the
+   * option of returning the value synchronously, which is useful in cases
+   * where it cannot be determined if the effect is synchronous or asynchronous
+   * until the effect is actually executed. The effect also has the option of
+   * returning a canceler, which will be used by the runtime to cancel the
+   * asynchronous effect if the fiber executing the effect is interrupted.
+   */
+  final def effectAsyncInterrupt[A](
+    register: (ZIO[UioR, UioE, A] => Unit) => Either[Canceler, ZIO[UioR, UioE, A]]
+  ): ZIO[UioR, UioE, A] =
+    ZIO.effectAsyncInterrupt(register)
+
+  /** Submerges the error case of an `Either` into the `ZIO`. The inverse
+   * operation of `IO.either`.
+   */
+  final def absolve[A](v: ZIO[UioR, UioE, Either[UioE, A]]): ZIO[UioR, UioE, A] =
+    ZIO.absolve(v)
+
+  /** The inverse operation `IO.sandboxed`
+   *
+   * Terminates with exceptions on the `Left` side of the `Either` error, if it
+   * exists. Otherwise extracts the contained `IO[E, A]`
+   */
+  final def unsandbox[A](v: ZIO[UioR, Cause[UioE], A]): ZIO[UioR, UioE, A] = ZIO.unsandbox(v)
+
+  /** Returns the identity effectful function, which performs no effects
+   */
+  final def identity: ZIO[UioR, Nothing, UioR] = ZIO.identity
+
+  /** Returns an effectful function that merely swaps the elements in a `Tuple2`.
+   *
+   *  Q: Leave Implicit?
+   */
+  final def swap[A, B](implicit ev: UioR <:< (A, B)): ZIO[UioR, UioE, (B, A)] =
+    ZIO.swap
+
+  /** Returns an effectful function that extracts out the first element of a
+   * tuple.
+   */
+  final def _1[A, B](implicit ev: UioR <:< (A, B)): ZIO[UioR, UioE, A] = ZIO._1
+
+  /** Returns an effectful function that extracts out the second element of a
+   * tuple.
+   */
+  final def _2[A, B](implicit ev: UioR <:< (A, B)): ZIO[UioR, UioE, B] = ZIO._2
```

Comment:
Yes, both these two must be deleted, not just in `UIO.scala`, but also in `Task.scala`, because they assume / require that `R` be some type other than `Any`.